azkaban-common/src/main/java/azkaban/sla/SlaOption.java

```diff
@@ -17,191 +17,266 @@
-  public static final String TYPE_JOB_SUCCEED = "JobSucceed";
-
-  public static final String INFO_DURATION = "Duration";
-  public static final String INFO_FLOW_NAME = "FlowName";
-  public static final String INFO_JOB_NAME = "JobName";
-  public static final String INFO_EMAIL_LIST = "EmailList";
-
-  // always alert
-  public static final String ALERT_TYPE = "SlaAlertType";
+  public static final String ALERT_TYPE_EMAIL = "email";
   public static final String ACTION_CANCEL_FLOW = "SlaCancelFlow";
   public static final String ACTION_ALERT = "SlaAlert";
   public static final String ACTION_KILL_JOB = "SlaKillJob";
+
+  public static final String WEB_ID = "id";
+  public static final String WEB_DURATION = "duration";
+  public static final String WEB_STATUS = "rule";
+  public static final String WEB_ACTIONS = "actions";
+  public static final String WEB_ACTION_EMAIL = "EMAIL";
+  public static final String WEB_ACTION_KILL = "KILL";
+
   private static final DateTimeFormatter fmt = DateTimeFormat
       .forPattern("MM/dd, YYYY HH:mm");
 
-  private String type;
-  private Map<String, Object> info;
-  private List<String> actions;
+  final private SlaType type;
+  final private String flowName;
+  final private String jobName;
+  final private String duration;
+  final private boolean alert;
+  final private boolean kill;
+  final private List<String> emails;
 
-  public SlaOption(final String type, final List<String> actions, final Map<String, Object> info) {
+  /**
+   * Constructor.
+   *
+   * @param type the SLA type.
+   * @param flowName The name of the flow.
+   * @param jobName The name of the job, if the SLA is for a job.
+   * @param duration The duration (time to wait before the SLA would take effect).
+   * @param alert if the user should be alerted for the SLA.
+   * @param kill if the job or flow should be killed or canceled for the SLA.
+   * @param emails list of emails to send an alert to, for the SLA.
+   */
+  public SlaOption(final SlaType type,
+      String flowName, String jobName, String duration, boolean alert, boolean kill,
+      List<String> emails) {
     this.type = type;
-    this.info = info;
-    this.actions = actions;
-  }
-
-  public static List<SlaOption> getJobLevelSLAOptions(final ExecutableFlow flow) {
-    final Set<String> jobLevelSLAs = new HashSet<>(
-        Arrays.asList(SlaOption.TYPE_JOB_FINISH, SlaOption.TYPE_JOB_SUCCEED));
-    return flow.getSlaOptions().stream()
-        .filter(slaOption -> jobLevelSLAs.contains(slaOption.getType()))
-        .collect(Collectors.toList());
-  }
-
-  public static List<SlaOption> getFlowLevelSLAOptions(final ExecutableFlow flow) {
-    final Set<String> flowLevelSLAs = new HashSet<>(
-        Arrays.asList(SlaOption.TYPE_FLOW_FINISH, SlaOption.TYPE_FLOW_SUCCEED));
-    return flow.getSlaOptions().stream()
-        .filter(slaOption -> flowLevelSLAs.contains(slaOption.getType()))
-        .collect(Collectors.toList());
+    this.flowName = flowName;
```

Comment:
consider assertion to guarantee the valid state of the object? e.g  SLAoption should have at least one action(kill or alert)

Reply:
Yes.